src/v/serde/serde.h

```diff
@@ -344,6 +408,28 @@ void write(iobuf& out, T t) {
 
         write(out, t.is_ipv4());
         write(out, std::move(address_bytes));
+    } else if constexpr (is_chrono_duration<Type>) {
+        // We explicitly serialize it as ns to avoid any surprises like
+        // seastar updating underlying duration types without
+        // notice. See https://github.com/redpanda-data/redpanda/pull/5002
+        //
+        // Check for overflows/underflows.
+        // For ex: a millisecond and nanosecond use the same underlying
+        // type int64_t but converting from one to other can easily overflow,
+        // this is by design.
+        // Since we serialize with ns precision, there is a restriction of
+        // nanoseconds::max()'s equivalent on the duration to be serialized.
+        // On a typical platform which uses int64_t for 'rep', it roughly
+        // translates to ~292 years.
+        //
+        // If we detect an overflow, we will clamp it to maximum supported
+        // duration, which is nanosecond::max() and if there is an underflow,
+        // we clamp it to minimum supported duration which is nanosecond::min().
```

Comment:
Should we also mention that the clamping is temporary and currently there is intention to remove it in the future, throwing an exception in case of overflow?

Reply:
Oh, I didn't know it was meant to be a temporary thing.. @dotnwat thats what you suggested?

Reply:
it's hard for me to imagine ever wanting to throw on overflow compared to clamping for duration types. but i'm sure there is some case out there where it makes sense. we can always revisit later.